contracts/WorkspaceRegistry.sol

```diff
@@ -144,6 +146,34 @@ contract WorkspaceRegistry is
         emit WorkspaceMembersUpdated(_id, _members, _roles, _enabled, _emails, block.timestamp);
     }
 
+    /**
+     * @notice Create an invite link for someone to join with
+     * @param _id ID of workspace to create invite link for
+     * @param publicKeyAddress Generated public key address for the invite link
+     * (corresponding private key should be sent to invitee)
+     */
+    function createInviteLink(uint96 _id, address publicKeyAddress) external whenNotPaused onlyWorkspaceAdmin(_id) {
+        bytes32 apiFlag = _apiFlagForWorkspaceId(_id);
+        AnonAuthoriser(ANON_AUTHORISER_ADDRESS).generateAnonAuthorisation(publicKeyAddress, apiFlag);
+    }
+
+    /**
+     * @notice Join a workspace with an invite link
+     * @param _id ID of workspace to join
+     * @param inviter Address of the person that sent the invite
+     * Remaining params are of the signature to be sent to AnonAuthoriser
+     */
+    function joinViaInviteLink(
+        uint96 _id,
+        address inviter,
+        uint8 signatureV,
+        bytes32 signatureR,
+        bytes32 signatureS
+    ) external whenNotPaused {
+        bytes32 apiFlag = _apiFlagForWorkspaceId(_id);
+        AnonAuthoriser(ANON_AUTHORISER_ADDRESS).anonAuthorise(inviter, apiFlag, signatureV, signatureR, signatureS);
```

Comment:
what should happen after this - if the link is successfully verified?

Reply:
Can you verify what is the msg.sender on the anonAuthorise?

I fear it'll be the WorkspaceRegistry.sol contract address

https://github.com/questbook/anon-authoriser/blob/4de969719810ac840d2c9e9c8fbfeba275d40d34/contracts/anon-authoriser.sol#L39

Reply:
> what should happen after this - if the link is successfully verified?

Nothing, will add the user to the workspace then -- TODO